### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by tarasavage**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            format_linter_error(err)
+            for err in errors
+            if err["filename"] == file_path
```
Comment (ArsenPidhoretskyi) [inline app/main.py:16]: No need to check this

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+        "status": "failed" if errors else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(key, value)
+        for key, value in linter_report.items()
```
Comment (ArsenPidhoretskyi) [inline app/main.py:26]: key, value is not good naming in this case

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
+            format_linter_error(err)
+            for err in errors
+            if err["filename"] == file_path
+        ],
+        "path": file_path,
+        "status": "failed" if errors else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:23]: return list of what?

File: app/main.py [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:11]: errors: list of all?

**Review by Grayder0152 — APPROVED** [review]

**Pull request closed without merging**